http://developer.pidgin.im/viewmtn/revision/info/5ccc437b944e0675b75037409a01416e31650e7f
What was I thinking... The GtkIMHtmlSmileys in the smiley trees aren't owned
by the smiley trees, so NULLing a pointer in there is not a good idea...
By the time gtk_smiley_tree_destroy is called the smileys are gone.
This should fix #7345 and that "crash when changing smiley theme while there
are smileys in input boxes" bug someone reported on #pidgin
Closes #7345 

diff -urN pidgin-2.5.2.orig/pidgin/gtkimhtml.c pidgin-2.5.2/pidgin/gtkimhtml.c
--- pidgin-2.5.2.orig/pidgin/gtkimhtml.c	2008-09-28 20:43:24.000000000 -0400
+++ pidgin-2.5.2/pidgin/gtkimhtml.c	2008-11-21 14:36:42.000000000 -0500
@@ -348,9 +348,7 @@
 			g_string_free (t->values, TRUE);
 			g_free (t->children);
 		}
-		if (t && t->image) {
-			t->image->imhtml = NULL;
-		}
+		
 		g_free (t);
 	}
 }
